### PR TITLE
New blog exceptions

### PR DESCRIPTION
**Pull request by jmglov**
Please answer the following questions and leave the below in as part of your PR.

- [x] This PR corresponds to an [issue with a clear problem statement](https://github.com/babashka/babashka/blob/master/doc/dev.md#start-with-an-issue-before-writing-code):
  #37 
  #38 

- [x] ~I have updated the [CHANGELOG.md](https://github.com/babashka/babashka/blob/master/CHANGELOG.md) file with a description of the addressed issue.~
  N/A because this is a bugfix

### CODE REVIEW

File: src/quickblog/api.clj [hunk 1]
```diff
@@ -437,11 +437,12 @@
          :as opts}
         (-> opts apply-default-opts lib/refresh-cache)]
     (when (empty? (:posts opts))
-      (if (fs/exists? posts-file)
-        (println (format "Run `bb migrate` to move metadata from `%s` to post files"
-                         posts-file))
-        (println "No posts found; run `bb new` to create one"))
-      (System/exit 1))
+      (throw (ex-info
```
Comment (borkdude) [inline src/quickblog/api.clj:440]: Do we really have to throw an exception when no posts are yet there? I don't think so ;)

Comment (borkdude) [reply src/quickblog/api.clj:440]: Just printing the message seems fine to me, to `*err*`.

Rest of the PR looks fine to me.

Comment (jmglov) [reply src/quickblog/api.clj:440]: OK. I just did it this way so it wouldn't exit in the REPL, but I see your point and can change it a little to accomplish both goals. :)

Comment (jmglov) [reply src/quickblog/api.clj]: Fixed!

File: src/quickblog/api.clj [hunk 2]
```diff
@@ -602,13 +604,14 @@
...
-        (watch assets-dir
-               (fn [{:keys [path type]}]
-                 (println "Asset change detected:"
-                          (name type) (str path))
-                 (when (contains? #{:remove :rename} type)
-                   (let [file (fs/file assets-out-dir (fs/file-name path))]
-                     (println "Removing deleted asset:" (str file))
-                     (fs/delete-if-exists file)))
-                 (lib/copy-tree-modified assets-dir assets-out-dir))))))
+        (when (fs/exists? assets-dir)
```
Comment (borkdude) [inline src/quickblog/api.clj:607]: @jmglov Did it crash before this check?

Comment (jmglov) [reply src/quickblog/api.clj:607]: Yes. That's the issue reported in #38 .

**Comment by borkdude** [comment]
Thanks!

**Pull request merged**